### PR TITLE
Format similarity responses

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

Take the similarity response and transform it for consumption by the front end

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/services/goalServices/helpers.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const responsesForComparison = (g) => (g.responses || [])
```
Comment (thewatermethod) [inline src/services/goalServices/helpers.js:1]: pulled this out from the recipient service so that we can use the same logic in two places

**Comment by thewatermethod** [comment]
@nvms Created this services to transform what you have, can you look this over when you have a second?

**Comment by nvms** [comment]
> @nvms Created this services to transform what you have, can you look this over when you have a second?

I did a small refactor last night to fix some bugs and add tests - will likely need to pull/merge again (sorry)

File: src/services/goals.js [hunk 2]
```diff
@@ -2571,3 +2572,140 @@ export async function createMultiRecipientGoalsFromAdmin(data) {
...
+      ],
+      status: 'Active',
+    },
+    attributes: ['id', 'oldGrantId', 'status'],
+  });
+
+  const grantLookup = {};
+  grants.forEach((grant) => {
+    grantLookup[grant.id] = grant.id;
+    grantLookup[grant.oldGrantId] = grant.id;
```
Comment (GarrettEHill) [inline src/services/goals.js:2676]: this could be a null key, as not all grants have an oldGrantId

Comment (thewatermethod) [reply src/services/goals.js:2676]: that's ok, ```grantLookup[current.grantId]``` will be falsy still

**Pull request merged**